packages/react/src/lib/renderComponent.tsx

```diff
@@ -166,17 +166,30 @@ const renderComponent = <P extends {}>(
     logProviderMissingWarning()
   }
 
-  const { disableAnimations = false, renderer = null, rtl = false, theme = emptyTheme } =
-    context || {}
+  const {
+    disableAnimations = false,
+    renderer = null,
+    rtl = false,
+    theme = emptyTheme,
+    resolvedComponentVariables = {},
+  } = context || {}
 
   const ElementType = getElementType(props) as React.ReactType<P>
   const stateAndProps = { ...state, ...props }
 
-  // Resolve variables for this component, allow props.variables to override
-  const resolvedVariables: ComponentVariablesObject = mergeComponentVariables(
-    theme.componentVariables[displayName],
-    props.variables && withDebugId(props.variables, 'props.variables'),
-  )(theme.siteVariables)
+  // Resolve variables for this component, cache the result in provider
+  if (!(displayName in resolvedComponentVariables)) {
```

Comment:
```suggestion
  if (resolvedComponentVariables[displayName]) {
```

If there difference between this?

Reply:
yes, they are completely opposite :-P

other than that, yes:
![image](https://user-images.githubusercontent.com/9615899/67089864-edec5900-f1a8-11e9-847f-dccc885d5767.png)

but because of the `|| {}` on line 183, this is no longer necessary

Reply:
removed